mysql/datadog_checks/mysql/mysql.py

```diff
@@ -299,7 +304,14 @@ def _collect_metrics(self, db, tags, options, queries, max_custom_queries):
             elif slave_running.lower().strip() == 'off':
                 if not (slave_io_running is None and slave_sql_running is None):
                     if not slave_io_running and not slave_sql_running:
+                        self.log.debug("Neither Slave_IO_Running or Slave_SQL_Running are not ok")
                         slave_running_status = AgentCheck.CRITICAL
+                    elif not slave_io_running or not slave_sql_running:
+                        self.log.debug("Either Slave_IO_Running or Slave_SQL_Running are not ok")
+                        slave_running_status = AgentCheck.WARNING
```

Comment:
Set as warning for consistency with the 5.7 code

Reply:
I would posit that if this check is not OK then it is CRITICAL so that line 306 would be an OR, thus obviating the need for the WARNING block.  The same would be true for the 5.7 code.

Reply:
Hey, another PR got merged after this one simplifying the whole thing https://github.com/DataDog/integrations-core/pull/8401

Reply:
Hmm - the logic [here](https://github.com/DataDog/integrations-core/blob/e01d17d5353207cc005323d9f01b73ab28fb4f3e/mysql/datadog_checks/mysql/mysql.py#L335-L337) still creates only a WARNING.  From the perspective of running databases the net effect of either of those two conditions not being met is that replication is broken which is CRITICAL.  This is using two different pieces of data to check replication status which is good but my position is that **both** aren't needed to determine if replication is broken, either would do that and that it is not correct to require both.  Therefore I think that line [332](https://github.com/DataDog/integrations-core/blob/e01d17d5353207cc005323d9f01b73ab28fb4f3e/mysql/datadog_checks/mysql/mysql.py#L332) should more correctly be an OR.

Reply:
This situation was already a WARNING for some instances in previous versions of the check and changing it to always be CRITICAL could affect existing monitors. When defining a monitor for a Service check you can define the same reporting policy for both so you will be equally alerted.

Reply:
Yes - I am sure it may affect existing monitors as there are likely other cases in your customer base where replication is broken but they are currently unaware due to this not reporting critical.